pkgs/servers/sql/postgresql/default.nix

```diff
@@ -145,6 +145,8 @@ let
 
     postInstall =
       ''
+        moveToOutput "bin/pg_config" "$dev"
```

Comment:
This is not sufficient. `pg_config` will contain wrong `bindir` or `docdir` paths, compare the output of `result-dev/bin/pg_config`

```Makefile
BINDIR = /home/jtojnar/Projects/nixpkgs/result-dev/bin
DOCDIR = /home/jtojnar/Projects/nixpkgs/result-dev/share/doc/postgresql
HTMLDIR = /home/jtojnar/Projects/nixpkgs/result-dev/share/doc/postgresql
INCLUDEDIR = /home/jtojnar/Projects/nixpkgs/result-dev/include
PKGINCLUDEDIR = /home/jtojnar/Projects/nixpkgs/result-dev/include
INCLUDEDIR-SERVER = /home/jtojnar/Projects/nixpkgs/result-dev/include/server
LIBDIR = /nix/store/fzsc6kbfc321pjjyy2is7scj860v617m-postgresql-14.9-lib/lib
PKGLIBDIR = /nix/store/fzsc6kbfc321pjjyy2is7scj860v617m-postgresql-14.9-lib/lib
LOCALEDIR = /home/jtojnar/Projects/nixpkgs/result-dev/share/locale
MANDIR = /home/jtojnar/Projects/nixpkgs/result-dev/share/man
SHAREDIR = /home/jtojnar/Projects/nixpkgs/result-dev/share/postgresql
SYSCONFDIR = /etc/postgresql
PGXS = /nix/store/vpk3sicb6q8yqljdzkpmiizi4nmcy9rf-postgresql-14.9/lib/pgxs/src/makefiles/pgxs.mk
CONFIGURE =  '--prefix=/nix/store/vpk3sicb6q8yqljdzkpmiizi4nmcy9rf-postgresql-14.9' '--with-openssl' '--with-libxml' '--with-icu' '--sysconfdir=/etc' '--libdir=$(lib)/lib' '--with-system-tzdata=/nix/store/dbr96xib9c1d3yqi5iqccz7a89r3qw18-tzdata-2023c/share/zoneinfo' '--enable-debug' '--with-systemd' '--with-ossp-uuid' '--with-lz4' '--with-gssapi' 'CC=cc' 'CXX=g++' 'PKG_CONFIG=pkg-config' 'PKG_CONFIG_PATH=/nix/store/31aay9sp0p6c1szqp63a7y9d942wrv7i-zlib-1.2.13-dev/lib/pkgconfig:/nix/store/fzjsi5yqqrz8i5j1mgw51gkmbk6g2dr3-readline-8.2p1-dev/lib/pkgconfig:/nix/store/pizm2gq99ikisqxxdf9rrvjnp88h99fw-ncurses-6.4-dev/lib/pkgconfig:/nix/store/vw1x6s2aiy31zdalzcjcsah21dxm3w70-openssl-3.0.10-dev/lib/pkgconfig:/nix/store/fnyn4kq0zkp438mmbpdbxiv8fx4zvk90-libxml2-2.11.4-dev/lib/pkgconfig:/nix/store/fkl3h1b9r4y95x46nwr2v14wj3yfhr9b-icu4c-73.2-dev/lib/pkgconfig:/nix/store/07ss01p01z6ilxk1wncslwyv781mlxrb-lz4-1.9.4-dev/lib/pkgconfig:/nix/store/9aqr6q8mpx4gbmg3vnz2h8v87scrnlrg-systemd-253.6-dev/lib/pkgconfig:/nix/store/9aqr6q8mpx4gbmg3vnz2h8v87scrnlrg-systemd-253.6-dev/share/pkgconfig:/nix/store/15a1dmj86lvi303b1nm9pmxjgvqrk0hg-libkrb5-1.20.1-dev/lib/pkgconfig:/nix/store/5py1ma3gndcqyvmzpl31wqr42l0l8qhz-libossp-uuid-1.6.2/lib/pkgconfig'
CC = cc
CPPFLAGS = -I/nix/store/fkl3h1b9r4y95x46nwr2v14wj3yfhr9b-icu4c-73.2-dev/include -D_GNU_SOURCE -I/nix/store/fnyn4kq0zkp438mmbpdbxiv8fx4zvk90-libxml2-2.11.4-dev/include/libxml2 -I/nix/store/07ss01p01z6ilxk1wncslwyv781mlxrb-lz4-1.9.4-dev/include
CFLAGS = -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wimplicit-fallthrough=3 -Wcast-function-type -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -g -O2
CFLAGS_SL = -fPIC
LDFLAGS = -L/nix/store/57ilg207pwyynwbnfggwdvgy6yjpg25a-libxml2-2.11.4/lib -L/nix/store/w2rhhc8f9r7wvhf6my8khzjl86jxz6bd-lz4-1.9.4/lib -Wl,--as-needed -Wl,-rpath,'/nix/store/fzsc6kbfc321pjjyy2is7scj860v617m-postgresql-14.9-lib/lib',--enable-new-dtags
LDFLAGS_EX = 
LDFLAGS_SL = 
LIBS = -lpgcommon -lpgport -llz4 -lxml2 -lssl -lcrypto -lgssapi_krb5 -lz -lreadline -lm 
VERSION = PostgreSQL 14.9
```

with one from #179962:

```Makefile
BINDIR = /nix/store/0rqhcqzhwra1pazc6bmkhd98gaf65hd5-postgresql-14.9/bin
DOCDIR = /nix/store/pklvxyyr2wzc99k7mq4mf94djv6nmmdr-postgresql-14.9-doc/share/doc/postgresql
HTMLDIR = /nix/store/pklvxyyr2wzc99k7mq4mf94djv6nmmdr-postgresql-14.9-doc/share/doc/postgresql
INCLUDEDIR = /dev/null/include
PKGINCLUDEDIR = /dev/null/include
INCLUDEDIR-SERVER = /dev/null/include/server
LIBDIR = /nix/store/vfb7fxx6l1pyj304pahxzq9934k42sgm-postgresql-14.9-lib/lib
PKGLIBDIR = /nix/store/vfb7fxx6l1pyj304pahxzq9934k42sgm-postgresql-14.9-lib/lib
LOCALEDIR = /nix/store/vfb7fxx6l1pyj304pahxzq9934k42sgm-postgresql-14.9-lib/share/locale
MANDIR = /nix/store/6x45mqhkjy3i6gfm5s0flmy2y3xjfy9x-postgresql-14.9-man/share/man
SHAREDIR = /nix/store/0rqhcqzhwra1pazc6bmkhd98gaf65hd5-postgresql-14.9/share
SYSCONFDIR = /etc/postgresql
PGXS = /nix/store/yl8n110sgfjpk4sbj3x8482h1m3p9rv5-postgresql-14.9-dev/lib/pgxs/src/makefiles/pgxs.mk
CONFIGURE =  '--prefix=/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-postgresql-14.9' '--bindir=/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-postgresql-14.9/bin' '--sbindir=/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-postgresql-14.9/sbin' '--includedir=/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-postgresql-14.9-dev/include' '--oldincludedir=/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-postgresql-14.9-dev/include' '--mandir=/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-postgresql-14.9-man/share/man' '--infodir=/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-postgresql-14.9/share/info' '--docdir=/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-postgresql-14.9-doc/share/doc/postgresql' '--libdir=/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-postgresql-14.9-lib/lib' '--libexecdir=/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-postgresql-14.9-lib/libexec' '--localedir=/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-postgresql-14.9-lib/share/locale' '--with-openssl' '--with-libxml' '--with-icu' '--sysconfdir=/etc' '--with-system-tzdata=/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-tzdata-2023c/share/zoneinfo' '--enable-debug' '--with-systemd' '--with-ossp-uuid' '--with-lz4' '--with-gssapi' 'CC=cc' 'CXX=g++' 'PKG_CONFIG=pkg-config' 'PKG_CONFIG_PATH=/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-file-5.45-dev/lib/pkgconfig:/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-libxcrypt-4.4.36/lib/pkgconfig:/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-libxslt-1.1.38-dev/lib/pkgconfig:/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-zlib-1.2.13-dev/lib/pkgconfig:/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-readline-8.2p1-dev/lib/pkgconfig:/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-ncurses-6.4-dev/lib/pkgconfig:/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-openssl-3.0.10-dev/lib/pkgconfig:/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-libxml2-2.11.4-dev/lib/pkgconfig:/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-icu4c-73.2-dev/lib/pkgconfig:/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-lz4-1.9.4-dev/lib/pkgconfig:/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-systemd-253.6-dev/lib/pkgconfig:/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-systemd-253.6-dev/share/pkgconfig:/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-libkrb5-1.20.1-dev/lib/pkgconfig:/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-libossp-uuid-1.6.2/lib/pkgconfig'
CC = cc
CPPFLAGS = -I/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-icu4c-73.2-dev/include -D_GNU_SOURCE -I/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-libxml2-2.11.4-dev/include/libxml2 -I/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-lz4-1.9.4-dev/include
CFLAGS = -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wimplicit-fallthrough=3 -Wcast-function-type -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -g -O2
CFLAGS_SL = -fPIC
LDFLAGS = -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-libxml2-2.11.4/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-lz4-1.9.4/lib -Wl,--as-needed -Wl,-rpath,/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-postgresql-14.9-lib/lib,--enable-new-dtags
LDFLAGS_EX = 
LDFLAGS_SL = 
LIBS = -lpgcommon -lpgport -llz4 -lxml2 -lssl -lcrypto -lgssapi_krb5 -lz -lreadline -lm 
VERSION = PostgreSQL 14.9
```

On the other hand, your PR handles the reverse dependencies, which I did not want to handle yet.